dbt_sugar/core/task/doc.py

```diff
@@ -56,10 +61,126 @@ def run(self) -> int:
+
+        Args:
+            model_name (str): model name to find in the dbt project.
+
+        Returns:
+            Tuple[Optional[Path], bool]: Optional path of the sql model if found
+            and boolean indicating whether the schema.yml exists.
+        """
+        for root, _, files in os.walk(self.repository_path):
+            files = [f for f in files if not re.match(EXCLUDE_TARGET_FILES_PATTERN, f)]
+            for file in files:
+                if file == f"{model_name}.sql":
+                    path_file = Path(os.path.join(root, "schema.yml"))
+                    if path_file.is_file():
+                        return path_file, True
+                    else:
+                        return path_file, False
+        return None, False
+
+    def orchestrate_model_documentation(self, model_name: str, columns_sql: List[str]) -> int:
+        """
+        Orchestrator to fully document a model will:
 
-        self.process_model(model, columns_sql)
+            - Create or update the columns found the database table into the schema.yml.
+            - Gives the user the posibility to change the model description.
+            - Gives the user the posibility to document the undocumented columns.
+            - Updates all the new columns definitions in all dbt.
+
+        Args:
+            model_name (str): Name of the model to document.
+            columns_sql (List[str]): Columns names that the model have in the database.
+
+        Returns:
+            int: with the status of the execution. 1 for fail, and 0 for ok!
+        """
+        content = None
+        path, exists = self.find_model_in_dbt(model_name)
+        if not path:
+            logger.info("Not able to find the model in the repository.")
+            return 1
+        if exists:
+            content = open_yaml(path)
+        content = self.process_model(content, model_name, columns_sql)
+        content = self.change_model_description(content, model_name)
+        save_yaml(path, content)
+
+        not_documented_columns = self.get_not_documented_columns(content, model_name)
+        self.document_columns(not_documented_columns)
+
+        self.update_column_descriptions(self.columns_to_update)
         return 0
 
+    def document_columns(self, columns: Dict[str, str]) -> None:
+        """
+        Method to document the columns from a model.
+
+        Will ask the user which columns they want to document and collect the new definition.
+
+        Args:
+            columns (Dict[str, str]): Dict of columns with the column name as the key
+            and the column description to populate schema.yml as the value.
+        """
+        columns_names = list(columns.keys())
+        for i in range(0, len(columns_names), NUMBER_COLUMNS_TO_PRINT_PER_ITERACTION):
+            final_index = i + NUMBER_COLUMNS_TO_PRINT_PER_ITERACTION
+            undocumented_columns_payload: List[Mapping[str, Any]] = [
```

Comment:
Oh that's kinda cool! That would ensure that we don't end up printing 100 columns all at once in the CLI correct?